client/src/components/SitterProfile/sendRequestCard/SendRequestDateTimePicker/SendRequestDateTimePicker.tsx

```diff
@@ -0,0 +1,155 @@
+import { Box, InputAdornment, TextField, Select, MenuItem, IconButton } from '@material-ui/core';
+import useStyles from './useStyles';
+import DateRangeIcon from '@material-ui/icons/DateRange';
+import { useSitters } from '../../../../context/useSitterContext';
+import DatePicker from 'react-multi-date-picker';
+import { useRef, useState, useEffect, useCallback, ChangeEvent, ReactNode } from 'react';
+import DateObject from 'react-date-object';
+import { useRequest } from '../../../../context/useRequestContext';
+
+interface availability {
+  [id: string]: { start: DateObject; end: DateObject };
+}
+
+interface props {
+  type: string;
+}
+
+export default function SendRequestDateTimePicker({ type }: props): JSX.Element {
+  const classes = useStyles();
+  const { requestADate, requestedStart } = useRequest();
+  const { selectedSitter } = useSitters();
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const DatePickerRef = useRef<any | null>();
+  const [availability, setAvailability] = useState<availability | null | undefined>(null);
+  const [selectedDate, setSelectedDate] = useState<DateObject | null | undefined>(null);
+  const [selectedTime, setSelectedTime] = useState<string | null | undefined>('');
+  const [availabileTimes, setAvailableTimes] = useState<string[] | null | undefined>(null);
```

Comment:
Great work designing the date pickers to only reflect sitter's available time slots :)

Reply:
Thank you